packages/nimble-components/src/table/specs/README.md

```diff
@@ -1,78 +1,272 @@
+#### Testing
 
-Pros
+As the performance for a table is of critical importance, we should formulate a deliberate approach to verifying the performance meets our expectations. For now, I recommend a series of manual tests, but ultimately it would be good to automate this process which I think we should cover in a separate HLD.
 
--   Architecture allows visualization to work with a dataset in the millions of rows without performance impact.
--   Plugin system allowing for the creation of various views that can be swapped out to visualize the underlying data.
--   Well tested
+### Dependencies
 
-Cons
+-   [TanStack Table](https://github.com/tanstack/table)
+-   [TanStack Virtual](https://github.com/TanStack/virtual)
 
--   Lack of support for hierarchical data
-    -   We could support hierarchical data by writing custom state management logic in the table component itself for features like grouping, sorting, and row selection. Some of which negates the performance benefits we would otherwise see for free.
--   Higher cost to implement a solution for, and more code to test/maintain.
+These dependencies are MIT licensed, and have no dependencies of their own, so security concerns should be at a minimum. When building a TS app that does not include the `nimble-table` the TanStack libraries are not included in the build output.
 
-Ultimately we decided against using Perspective as the general purpose table as one of its most appealing traits, its performance, would in many situations be neutralized by the implementation needed to support use cases like row selection and hierarchical data.
+### Test Plan
 
-We do see future potential for leveraging Perspective for more targeted data analysis use-cases like:
+Intend to test completely with unit tests and Chromatic visual tests.
 
--   as a parametric data analysis engine with multi-visualization cross-filtering
--   client side visualization and analysis of large data sets (100k - 10M)
--   interactive analysis of massive server-side data sets (10M - 1B+).
+We also should attempt to create unit tests to cover the TanStack capabilities we will be leveraging, ideally by submitting them to the TanStack repo. Note that TanStack has already implemented much of the needed tests, except that they are targeting a React environment. The work should largely be a porting effort.
 
-There likely will be ways to leverage many of the general purpose table features (API, column configuration, styling, etc) in a theoretical future Perspective table if clients request that level of performance capability.
+### Documentation
 
-### Tabulator
+Storybooks will be added to document/showcase the various features and APIs.
 
-[Tabulator](https://tabulator.info/) is a full-fledged table web component, offering a wide range of features that are well aligned with our needs.
+### Open Issues
```

Comment:
Since we don't plan to resolve this in the initial submission, I'd vote to move it to the corresponding section of the doc and just mark it as an HLD that needs to be written (similar to what you did for all the API HLDs). I could see this going in the Accessibility section.

Reply:
Done.